server/src/main/java/org/opensearch/search/DocValueFormat.java

```diff
@@ -260,7 +265,14 @@ public String getWriteableName() {
 
         @Override
         public void writeTo(StreamOutput out) throws IOException {
-            out.writeString(formatter.pattern());
+            if (out.getVersion().before(Version.V_3_0_0) && formatter.equals(DateFieldMapper.DEFAULT_DATE_TIME_FORMATTER)) {
+                out.writeString(DateFieldMapper.LEGACY_DEFAULT_DATE_TIME_FORMATTER.pattern());
```

Comment:
Didn't understand the use case around `LEGACY_DEFAULT_DATE_TIME_FORMATTER`. Can you add a code comment on why its needed?

Reply:
`LEGACY_DEFAULT_DATE_TIME_FORMATTER` is needed to maintain backward compatibility b/w current version 3.0.0 and last minor version 2.11 in a mixed cluster. Let me explain it through an upgrade scenario where a cluster contain 2 nodes, 1 is on version 3.0. and another on 2.11. The 3.0 node will have the `printFormat` changes but 2.11 node won't have it(considering backporting has not been done till yet). When an aggregation query will come on 3.0 node it'll try to send the datetime formatter to 2.11 node for any printing to be done for the aggregation buckets but it won't be able to send the `printFormat` because the node on the other side of stream doesn't have the required changes to receive it. Therefore, in the case where default formatter is being used in aggregation query the new default format pattern with `strict_datetime_no_millis` will be sent to the 2.11 node and the 2.11 node will be print aggregation buckets according to this new default format. Because the new default format has `strict_datetime_no_millis` as the first formatter everything will be printed according to thius first formatter which is wrong and a breaking change for clients. Therefore, to mitigate this issue I kept. the old formatter if the node on the other side of a stream is on a lower version. When this change will be backported to 2.11 I can remove this from 3.0. This check will still be there in 2.11 backporting with the version check change to less than 2.11

Reply:
@CaptainDredge I don't understand the need to have separate serialization for default LEGACY_DEFAULT_DATE_TIME_FORMATTER and I don't see where this constant is being used used. It looks like just writing `formatter.pattern()` in all cases would be sufficient (in case of LEGACY_DEFAULT_DATE_TIME_FORMATTER, we do exactly that anyway).

```
out.writeString(formatter.pattern());
```

Reply:
Sure. Please add a code comment as well.